test/dependency_source/test_requirement.py

```diff
@@ -94,3 +96,57 @@ def test_requirement_source_duplicate_dependencies(monkeypatch):
     # If the dependency list has duplicates, then converting to a set will reduce the length of the
     # collection
     assert len(specs) == len(set(specs))
+
+
+# The path types have read-only attributes that can't be monkeypatched. Therefore, we'll need to
+# have a custom mock type that looks like a path so that we can deliver our mocked file.
+class MockFile:
+    def __init__(self) -> None:
+        self.contents = str()
+
+    def write(self, line: str) -> None:
+        self.contents += line + "\n"
+
+    def __enter__(self) -> "MockFile":
+        return self
+
+    def __exit__(self, _exc_type, _exc_value, _exc_traceback) -> None:
+        pass
+
+
+class MockPath:
+    def __init__(self, path: str) -> None:
+        self.path = path
+        self.mock_file = MockFile()
+
+    def open(self, *_args, **_kwargs) -> MockFile:
+        return self.mock_file
+
+    def __str__(self) -> str:
+        return self.path
+
+    def __fspath__(self) -> str:
+        return self.path
+
+
+def test_requirement_source_fix(monkeypatch):
```

Comment:
This was already troublesome enough, but I'll need to figure out a way to mock out the recovery parts.

Instead of continuing down this path, I'm thinking I might just have a helper in `conftest.py` to create a real requirements file in the form of a temporary file and inject text into it. That way, I don't need to mock anything.

Reply:
Yeah, having a real requirements file as a test fixture seems like a more reliable/less painful approach. If we have to mock anything at all for these tests, we should probably limit it to just the fix directives and not the I/O or FS ops.

Reply:
(i.e., the test should be "pass a tempfile with some contents in, and assert that those contents have changed as specified by a phony fix directive")